common/fixation_entry.go

```diff
@@ -60,15 +61,60 @@ func GetEntry(ctx sdk.Context, storeKey sdk.StoreKey, entryKeyPrefix string, cdc
 	}
 
 	cdc.MustUnmarshal(b, &val)
+
+	switch referenceAction {
+	case types.ADD_REFERENCE:
+		val.References += 1
+		SetEntry(ctx, storeKey, entryKeyPrefix, cdc, val)
+	case types.SUB_REFERENCE:
+		if val.GetReferences() > 0 {
+			val.References -= 1
+			SetEntry(ctx, storeKey, entryKeyPrefix, cdc, val)
+		}
+	case types.DO_NOTHING:
+	}
+
 	return val, true
 }
 
+// Helper function to extract the original index from an older version index (for example: myIndex_2 -> myIndex)
+func extractOriginalIndexFromOlderVersionIndex(index string) (originalIndexWithoutVersionNumSuffix string) {
+	// check that the index's length is non-zero
+	if len(index) == 0 {
+		return ""
+	}
+
+	// if the index doesn't contain "_", it's not an older version index (as defined in this code) so there's nothing to do
+	if !strings.Contains(index, "_") {
+		return index
+	}
+
+	// trim the index string from the start to the last "_"
+	if i := strings.LastIndex(index, "_"); i > 0 {
+		index = index[:i]
+	}
+
+	return index
+}
+
 // Remove entry with full index. Full index is the entry index + version num suffix (like "bundle1_0")
-func RemoveEntry(ctx sdk.Context, storeKey sdk.StoreKey, entryKeyPrefix string, entryIndex string) {
+func RemoveEntry(ctx sdk.Context, storeKey sdk.StoreKey, cdc codec.BinaryCodec, entryKeyPrefix string, entryIndex string) {
```

Comment:
maybe like the get, that receives key and block and than delete this one will be better. we dont want to rely on string manipulations

Reply:
done